rust/src/grpc/mod.rs

```diff
@@ -15,13 +16,27 @@ use interceptor::AuthInterceptor;
 pub type SiftChannel = InterceptedService<Channel, AuthInterceptor>;
 
 /// Uses `channel_config` to initialize a lazy channel that will only establish a connection
-/// after first-use. This can only be used within a [`Tokio 1.x runtime`].
+/// after first-use. This can only be used within a [`Tokio 1.x runtime`]. The returned [`SiftChannel`] is
+/// a gRPC Channel that is pre-configured with an authorization interceptor as well as HTTP2
+/// keep-alive. If these settings don't suit your needs, prefer to use a custom [`tonic endpoint`].
 ///
 /// [`Tokio 1.x runtime`]: https://docs.rs/tokio/latest/tokio/
+/// [`tonic endpoint`]: https://docs.rs/tonic/latest/tonic/transport/channel/struct.Endpoint.html
 pub fn use_sift_channel(channel_config: SiftChannelConfig) -> Result<SiftChannel> {
     let SiftChannelConfig { uri, apikey } = channel_config;
 
+    let crate_name = env!("CARGO_PKG_NAME");
+    let crate_version = env!("CARGO_PKG_VERSION");
+    let user_agent = format!("{crate_name}/{crate_version}");
+
     let channel = Endpoint::from_shared(uri)
+        .map(|conn| {
+            conn.keep_alive_while_idle(true)
```

Comment:
Do we eventually want to allow users to pass in their own keepalive config similar to how it's set up on the python side?

Reply:
I will err on the side of no, for the time being because I don't think it's something the vast majority of users will want to think about on their own - they just want something that works well out of the box. Ultimately if they want to get really granular with their keepalive settings they are the ability to create their own custom grpc channel and use that instead

Reply:
Sounds good!